### DISCUSSION TITLE
Update node promise result throw error but update result is right

### DISCUSSION THREAD

**Discussion by youjiaqi421** [xyflow/xyflow#3589] [Q&A]
1.Click the button and the server returns the result to update the node but it's will throw error
<img width="1467" alt="WeChat069866e086159206029d209c74b9221e" src="https://github.com/xyflow/xyflow/assets/59857905/14bf7719-3913-4795-b80d-3d74fe94b41f">
2.It is good if you use the following method

<img width="565" alt="WeChat0aef3965a18f18701d9a58b4e315d7ee" src="https://github.com/xyflow/xyflow/assets/59857905/09aa842b-8760-4c95-beb9-3d61e9377656">

help help

**Comment by bcakmakoglu** [comment]
What error do you receive? Can you elaborate a bit more on that?
